### PR TITLE
GP based generic batch-UCB agent

### PR DESCRIPTION
**Pull request by MuratAykol-TRI**
This PR implements a new batch-mode UCB-based acquisition approach from Desautels et al. Journal of Machine Learning Research 15 (2014) 4053-4103 in agent `GPBatchUCB`. 
- This agent composes a batch in an iterative way that avoids over-exploration in a given batch due to clustered solutions. It can also automatically adjust the uncertainty mixing parameter alpha. Kernels and other hyperparameters can be customized for a given application.
- This agent also covers both the strictly sequential (batch size=1) as well as "naive" batch composition techniques.
- Internally tested with various datasets.

`GPBatchUCB` can serve as the default BO-agent option in camd for a lot of optimization problems where dataset sizes allow the use of GPs.

### CODE REVIEW

**Comment by chirranjeevigopal-TRI** [comment]
Thanks for coding it up. Look forward to the reviewing this -- will get to it tonight.

File: camd/agent/generic.py [hunk 1]
```diff
@@ -60,3 +62,146 @@ def get_hypotheses(self, candidate_data, seed_data=None):
...
+                    print("- alpha.{}: ".format(i), alpha)
+                else:
+                    alpha = self.alpha
+
+                t_pred += unc * alpha
+                s = np.argmax(t_pred)
+
+                name = r_candidates.index.tolist()[s]
+                batch.append(name)
+                r_seed = r_seed.append(r_candidates.loc[name])
```
Comment (chirranjeevigopal-TRI) [inline camd/agent/generic.py:181]: Inferred value for the candidate being presumed to be 'measured' value and using that to obtain other candidates for the batch seems to be clutch for the 'auto' mode. Can you add some annotation reg that?

File: camd/agent/generic.py [hunk 2]
```diff
@@ -60,3 +62,146 @@ def get_hypotheses(self, candidate_data, seed_data=None):
...
+        self.seed_data = seed_data
+
+        fb_start = max(len(self.seed_data), 1)
+
+        if self.kernel is None:
+            self.kernel = GPy.kern.RBF(input_dim=self.candidate_data.shape[1])
+
+        X_seed = seed_data.drop(columns=["target"], axis=1)
+        y_seed = seed_data["target"].to_numpy().reshape(-1, 1)
+
```
Comment (chirranjeevigopal-TRI) [inline camd/agent/generic.py:127]: Does this code block run if `seed_data` is None ?

**Comment by MuratAykol-TRI** [comment]
@chirranjeevigopal-TRI thanks! I'll look at the points you note, and @JosephMontoya-TRI will help fix the integration issues before we merge.

**Comment by MuratAykol-TRI** [comment]
@JosephMontoya-TRI do you have any ideas why this is still failing ? I pulled the changes from master and everything looks identical related to build.

**Comment by JosephMontoya-TRI** [comment]
Hmm... it looks like there's an issue with teh branch name in deployment, let's just merge for now and I think it'll build correctly, I can fix before release.

**Pull request merged**